### DISCUSSION TITLE
Wrong time reported under entity nodes

### DISCUSSION THREAD

**Discussion by rosariopomillo** [zachowj/node-red-contrib-home-assistant-websocket#446] [Q&A]
Hi all, 
first of all thank you, all contributors to this fantastic world!
I have to report a very little but annoyig (possible) bug: at midnight the time reported under those nodes which displays time, the hour is reported as 24 instead o 0. I don't know if this normal, but it is totally unfamiliar to me; I live in Italy where 24 hours raprentation of time is wideky used (as in the rest of continental EU I suppose).
Than, 1st question: is it a bug? (see attached picture), and if yes
2) it depends on node-red or in hassio?
3) is it possible the user may manage this behaviour?

thank you in advance,
bye
Rosario
![Cattura](https://user-images.githubusercontent.com/44002283/126014568-e45749fa-fd15-4bee-8e86-b0d485331364.PNG)

**Comment by zachowj** [comment]
It has to do with a quirk with `toLocaleDateString` for the locale code `en-US` shows `24` instead of `00`. I could use the locale defined in the NR settings but I don't think it is exposed to nodes in the runtime.

https://github.com/zachowj/node-red-contrib-home-assistant-websocket/blob/56d063f1b0d320a8e75389dbbb275a5ddf562061/src/helpers/status.ts#L89

You can change the hard-coded value to your location to fix this issue.

Somewhere on the to-do list, is to make this dynamic.

**Reply by rosariopomillo** [reply]
Thanks a lot for your answer.
I have to admit I understood less than half you said. 
But now I know the issue is traced and soon or later it will be managed in a user friendly way.
Bye
Rosario
